### ISSUE TITLE
Update base branch of PR #2 once PR #1 is merged

### ISSUE THREAD

**Issue by fwouts** [Originate/git-town#1190]
First off, thanks a ton for building Git Town! It looks like it will solve some major workflow issues for our team of ~40 engineers.

I tested the following scenario:
1. `git town hack step-1`, make some changes, send out a GitHub PR with `git town new-pull-request`. This results in a PR #1 to merge `step-1` into `master`.
2. While `step-1` is being reviewed, I start working on `step-2` with `git town append step-2`.
3. `step-2` is ready for review before `step-1` is fully reviewed. I send out a PR with `git town new-pull-request`. This results in a PR #2 to merge `step-2` into `step-1`.
4. `step-1` is approved, so I merge it into master (squash & merge through GitHub's UI). However I cannot delete the branch `step-1` because it's referenced by PR #2.

I think it does make sense that (3) resulted in a PR with the base branch `step-1`, since it allows a reviewer to only see what has changed between `step-1` and `step-2`. However, once `step-1` was merged into master, I expected `git town sync` to automatically update PR #2, setting the base branch to `master`.

This seems like a minor annoyance, but there's a risk that developers will mistakenly merge PRs into the parent branch (which has already been merged) instead of merging into `master`.

Is this something that's already been considered? Could we potentially leverage GitHub's API to find out that the parent PR was merged, and update children PRs accordingly?

I'm keen to contribute to help address this issue, as long as you have some guidance on the approach to use. Thanks!

**Comment by kevgo** [comment]
Hey @fwouts, thanks for the kind words and for using Git Town! Your problem can be solved with the `git town ship` command. If you configure Git Town with your GitHub API token (`git config --add git-town.github-token <your token>`), it uses the Github API to perform the merge. This means the PR looks properly merged on GitHub, all the pending PR that have this branch as their base branch get updated, and the branch is deleted from Github (as well as the local dev machine).

We think that shipping from the command line is the best option for several reasons:
1. It makes sure all changes from the local dev machine are pushed to Github before merging.
1. It resolves merge conflicts before doing the actual merge.
1. The developer who coded the changes in the branch shows up in the Git history of the main branch.
1. If there are any merge conflicts before shipping the branch, the developer that coded it has to resolve them (instead of the person merging). They know their changes better than anybody else.
1. If there are problems after the merge, the developer who coded the offending  branch has to fix it, rather than the person merging. 
1. The shipped branch gets deleted from the developer's machine. If you merge branches via the GitHub UI, you have to run `git prune-branches` on the command line to remove them from your machine. If we have to go to the terminal after every merge done via the Github UI, why not run `git town ship` to begin with?

I hope this makes sense. Does this work for you?

**Comment by fwouts** [comment]
This is amazing. Your explanation makes sense.

May I suggest prompting for a GitHub API token the first time `git town` is used in a repo (when you detect that the remote is GitHub), so people don't miss out on this feature?

**Comment by kevgo** [comment]
That's a great idea! I have created #1191 to track it separately from the issue discussed here. Thanks for the feedback and suggestions! Closing this ticket since it seems resolved.

**Comment by kevgo** [comment]
FYI GitHub now automatically updates the base branch of proposals of child branches.
